Phase5/Search/Tokenizer.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System.Collections.Generic;
+using System.Text.RegularExpressions;
+
+
+namespace Phase5
+{
+    public class Tokenizer : ITokenizer
+    {
+        public HashSet<string> Tokenize(string doc)
+        {
+            var ans = new HashSet<string>();
+            var regex = new Regex("[a-zA-Z]+");
+            var matches = regex.Matches(doc);
+            foreach (Match match in matches)
+            {
+                var word = match.Value;
+                if (word.Length > 1)
+                    ans.Add(word.ToLower());

```

Comment:
it's better to make whole of content lower case at the beginning of method so you will have less overhead of calling this method in every iteration of foreach loop.